packages/protocol/contracts/identity/FederatedAttestations.sol

```diff
@@ -174,10 +251,122 @@ contract FederatedAttestations is
         trimmedIssuedOns[i] = issuedOns[i];
         trimmedSigners[i] = signers[i];
       }
-      return (trimmedAccounts, trimmedIssuedOns, trimmedSigners);
+      return (countsPerIssuer, trimmedAccounts, trimmedIssuedOns, trimmedSigners);
     } else {
-      return (accounts, issuedOns, signers);
+      return (countsPerIssuer, accounts, issuedOns, signers);
+    }
+  }
+
+  /**
+   * @notice Similar to lookupAttestations but returns all attestations for
+   *   an identifier produced by a list of issuers, regardless of whether
+   *   signers have been revoked or not
+   * @param identifier Hash of the identifier
+   * @param trustedIssuers Array of n issuers whose attestations will be included
+   * @return [0] Array of number of attestations returned per issuer
+   * @return [1 - 3] for m (== sum([0])) found attestations:
+   *         [
+   *           Array of m accounts,
+   *           Array of m issuedOns,
+   *           Array of m signers
+   *         ]; index corresponds to the same attestation
+   * @dev Adds attestation info to the arrays in order of provided trustedIssuers
+   * @dev Expectation that only one attestation exists per (identifier, issuer, account)
+   */
+  function lookupAllAttestations(bytes32 identifier, address[] calldata trustedIssuers)
+    external
+    view
+    returns (uint256[] memory, address[] memory, uint256[] memory, address[] memory)
+  {
+    uint256 totalAttestations;
+    uint256[] memory countsPerIssuer;
+
+    (totalAttestations, countsPerIssuer) = getTotalNumberOfAttestations(identifier, trustedIssuers);
+    address[] memory accounts = new address[](totalAttestations);
+    uint256[] memory issuedOns = new uint256[](totalAttestations);
+    address[] memory signers = new address[](totalAttestations);
+
+    OwnershipAttestation[] memory attestationsPerIssuer;
+    // Reset this and use as current index to get around stack-too-deep
+    totalAttestations = 0;
+
+    for (uint256 i = 0; i < trustedIssuers.length; i = i.add(1)) {
+      attestationsPerIssuer = identifierToAddresses[identifier][trustedIssuers[i]];
+      for (uint256 j = 0; j < attestationsPerIssuer.length; j = j.add(1)) {
```

Comment:
Nit: (don't worry about this if it feels like too big a change) If we already have counts per issuer, then maybe we could use that for our loop ? Just an idea, may not make sense

Reply:
i think i may be misunderstanding what you mean here, but I think it may be less readable to loop through that array? will leave this for now but feel free to re-raise on the main PR when you're back from OOO!